src/runner/mod.rs

```diff
@@ -0,0 +1,251 @@
+//! Runners are responsible for executing a test suite's examples.
+
+mod configuration;
+mod observer;
+
+pub use runner::configuration::*;
+pub use runner::observer::*;
+
+use std::panic;
+use std::sync::{Arc, Mutex};
+use std::borrow::Borrow;
+use std::cell::Cell;
+use std::process;
+use std::ops::{Deref, DerefMut};
+
+use rayon::prelude::*;
+
+use block::Block;
+use block::Suite;
+use block::Context;
+use block::Example;
+use report::{Report, BlockReport};
+use report::ContextReport;
+use report::SuiteReport;
+use report::ExampleReport;
+use visitor::TestSuiteVisitor;
+
+/// Runner for executing a test suite's examples.
+pub struct Runner {
+    configuration: configuration::Configuration,
+    observers: Vec<Arc<RunnerObserver>>,
+    should_exit: Mutex<Cell<bool>>,
+}
+
+impl Runner {
+    pub fn new(configuration: Configuration, observers: Vec<Arc<RunnerObserver>>) -> Runner {
+        Runner {
+            configuration: configuration,
+            observers: observers,
+            should_exit: Mutex::new(Cell::new(false)),
```

Comment:
why do you need a Mutex<Cell> ?

Reply:
I'm not sure we do. I tried lots of things and `Mutex` was the first to make rustc shut up. 😅🙈

Please feel free to improve this! I haven't done too much Rust with mutable state shared across threads. As such I probably chose the wrong types. It also feels way more complicated than it should be.